### ISSUE TITLE
Updating a polyline causes a crash

### ISSUE THREAD

**Issue by jenz0rw00t** [mapbox/mapbox-maps-android#973]
## Environment

- Android OS version: 8, 9 ,10, 11, 12
- Devices affected: Samsung: S8, S9, S10. Motorola: E7 plus, One zoom. Huawei: P30 pro, P30 lite, among others
- Maps SDK Version: 10.1.0

## Observed behavior and steps to reproduce

In the app I'm working on we draw a polyline behind the user based on the coordinates we get from FusedLocationProvider. Before updating to Maps SDK v.10 this was no issue, but now this is causing a crash for many of our users. I've tried both using PolylineAnnotationManager and LineLayer and GeoJsonSource to draw the lines but both ways cause a crash. I can't reproduce this crash on our own devices but we get crashlogs from our users. 

This is the stack trace:
```
Fatal Exception: java.util.ConcurrentModificationException
       at java.util.ArrayList$Itr.next(ArrayList.java:860)
       at com.mapbox.geojson.ListOfPointCoordinatesTypeAdapter.write(ListOfPointCoordinatesTypeAdapter.java:32)
       at com.mapbox.geojson.ListOfPointCoordinatesTypeAdapter.write(ListOfPointCoordinatesTypeAdapter.java:19)
       at com.mapbox.geojson.BaseGeometryTypeAdapter.writeCoordinateContainer(BaseGeometryTypeAdapter.java:75)
       at com.mapbox.geojson.LineString$GsonTypeAdapter.write(LineString.java:310)
       at com.mapbox.geojson.LineString$GsonTypeAdapter.write(LineString.java:302)
       at com.google.gson.Gson.toJson(Gson.java:704)
       at com.google.gson.Gson.toJson(Gson.java:683)
       at com.google.gson.Gson.toJson(Gson.java:638)
       at com.google.gson.Gson.toJson(Gson.java:618)
       at com.mapbox.geojson.LineString.toJson(LineString.java:235)
       at com.mapbox.maps.extension.style.utils.TypeUtilsKt.toValue(TypeUtils.kt:313)
       at com.mapbox.maps.extension.style.sources.generated.GeoJsonSource.toPropertyValue(GeoJsonSource.kt:324)
       at com.mapbox.maps.extension.style.sources.generated.GeoJsonSource.access$toPropertyValue(GeoJsonSource.kt:33)
       at com.mapbox.maps.extension.style.sources.generated.GeoJsonSource$applyGeoJsonData$$inlined$apply$lambda$1.run(GeoJsonSource.kt:337)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:255)
       at android.os.HandlerThread.run(HandlerThread.java:67)
```

How we update the polyline with GeoJsonSource:
```
val lineString = LineString.fromLngLats(points)
trackingSource.geometry(lineString)
```

How we update the polyline with PolylineAnnotationManager:
```
val lineString = LineString.fromLngLats(points)
trackingLine.geometry = lineString
filledLineManager.update(trackingLine)
```

## Expected behavior

Before updating to Maps v.10 this didn't cause a crash and back then we update the polyline coordinates in the same way (using LineManager and before that using LineLayer and GeoJsonSource ) and as frequent as now.

## Notes / preliminary analysis

I tried using PolylineAnnotationManager as well as LineLayer and GeoJsonSource but both ways caused the same crash.

**Comment by kiryldz** [comment]
@jenz0rw00t it's caused by async geojson parsing happening under the hood of the SDK. 

I'm not sure about precisely your case but in general `ConcurrentModificationException` was sometimes thrown and caught by our robo tests for our test app. Fix was relatively easy, see https://github.com/mapbox/mapbox-maps-android/pull/926.

Could you look try applying similar approach in you code?

**Comment by jenz0rw00t** [comment]
@kiryldz Thanks for the response! I've implemented a similar fix like the one from #926, I will keep an eye out to see if the crash is completely gone for our users since i couldn't reproduce it on any of our devices.

**Comment by kiryldz** [comment]
Closing this for now, feel free to re-open if issue persists!
